packages/rum/src/boot/recorderApi.ts

```diff
@@ -92,16 +92,23 @@ export function makeRecorderApi(startRecordingImpl: StartRecording): RecorderApi
             return
           }
 
-          const { stop: stopRecording } = startRecordingImpl(
+          const recording = startRecordingImpl(
             lifeCycle,
             initConfiguration.applicationId,
             configuration,
             session,
             parentContexts
           )
-          state = {
-            status: RecorderStatus.Started,
-            stopRecording,
+
+          if (!recording) {
+            state = {
+              status: RecorderStatus.Stopped,
+            }
+          } else {
+            state = {
+              status: RecorderStatus.Started,
+              stopRecording: recording.stop,
```

Comment:
Not sure if this is a standard of the SDK but I find this to be harder to grasp that it should be.
In my opinion`startRecordingImpl` should not return 
```ts
{ stop: () => void } | undefined
```
but something more explicit like that to track the status
```ts
{ stop: () => void, isSuccess: boolean }
```

Reply:
Your comment lead to a bit of discussion in the team. Basically, we don't want to introduce this kind of pattern right now, but we decided to try to start the worker a bit earlier so less functions can "fail" (see last commit). I think it improves things a bit. WDYT?